app/models/concerns/location_validations.rb

```diff
@@ -0,0 +1,24 @@
+require 'active_support/concern'
+
+module LocationValidations
+  extend ActiveSupport::Concern
+
+  included do |sti_klass|
+    location_key = sti_klass::LOCATION_ATTRIBUTE_KEY
+
+    sti_klass.define_method(location_key) do
+      details[location_key]
+    end
+
+    sti_klass.define_method("#{location_key}=") do |location_id|
+      details[location_key] = location_id
+    end
+
+    validates location_key, presence: true
+    validates_each location_key do |record, attr, value|
+      Location.find(value)
+    rescue ActiveRecord::RecordNotFound
+      record.errors.add(attr, "The location relationship you passed has an id that does not exist in our system. Please use an existing #{location_key.to_s.sub('_id', '')}")
```

Comment:
Currently for all other instances of `ActiveRecord::RecordNotFound` in the database, we fallback to this error on the API response: https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api_controller.rb#L103
To maintain consistency, we should not add a custom validation/error message here, but allow the API controller to handle that error as default

Reply:
I had debated this myself, actually. I ended up thinking I could give a clearer message to the user in the event of the value not existing. I think on balance the better solution is to be consistent/cause fewer questions about what's going on. It's actually cleaner code, too.